### PR TITLE
transition to github actions

### PR DESCRIPTION
**Pull request by grahamgower**

### CODE REVIEW

**Review by jeromekelleher — APPROVED** [review]

**Comment by grahamgower** [comment]
This is still quite broken, unfortunately. I'm trying to figure it out in a test repository elsewhere so as to avoid creating more noise here.

**Comment by grahamgower** [comment]
I'm fairly condfident this is doing the right thing now @jeromekelleher. It would be great if you could disable appveyor and travis now, as I've removed those config files in this PR. I've left circleci alone for now, because it's doing the codecov upload. We can move that to actions too, but I'd prefer to not look at this stuff again for a while. :)

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        id: cache-slim
+        uses: actions/cache@v2
+        env:
+          # Increase this to reset the cache if the key hasn't changed.
+          SLIM_CACHE_NUM: 0
+        with:
+          path: ~/slim
+          key: ${{ runner.os }}-slim-${{ env.SLIM_TAG }}-${{ env.SLIM_CACHE_NUM }}
+
+      - name: build SLiM
```
Comment (jeromekelleher) [inline .github/workflows/tests.yml:94]: Why are we building SLiM? Can we not use the [conda package](https://anaconda.org/conda-forge/slim)?

**Review by jeromekelleher — COMMENTED** [review]
LGTM - one question about SLiM though

**Comment by jeromekelleher** [comment]
I've disabled builds on Travis and Appveyor. I haven't deleted the configs yet, though, we'll want to loop back around to do that at some point.

**Comment by jeromekelleher** [comment]
Also: hooray! Thanks for getting this sorted @grahamgower - I know how painful getting this stuff up and running is.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (grahamgower) [reply .github/workflows/tests.yml:94]: Umm... because for some reason I didn't even think of that. Will fix!

**Comment by jeromekelleher** [comment]
Woohoo! I think there's a few stray `` env.SLIM_TAG ``s lying around that don't do anything now though?

**Comment by grahamgower** [comment]
Because we can't install slim from conda on windows (it doesn't exist), we can't just include `slim==3.4` in the conda requirements file. So there's a conditional statement in the `install dependencies` step that installs slim, and I've used the `env.SLIM_TAG` to provide the version string there. This version tag is also used for the cache key in the `cache conda` step (so if the slim version gets bumped, then the cache will be invalidated).

**Pull request merged**